backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2023 Volkswagen AG
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.puris.backend.stock.logic.adapter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Material;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Partner;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialPartnerRelationService;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialService;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.MaterialItemStock;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.ProductItemStock;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.ReportedMaterialItemStock;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.ReportedProductItemStock;
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@Slf4j
+public class ItemStockSammMapper {
+
+    @Autowired
+    private MaterialService materialService;
+    @Autowired
+    private MaterialPartnerRelationService mprService;
+
+    public ItemStockSAMM toItemStockSAMM(MaterialItemStock materialItemStock) {
+        ItemStockSAMM samm = new ItemStockSAMM();
+        samm.setDirection(DirectionCharacteristic.INBOUND);
+        samm.setPositions(new ArrayList<>());
+        samm.setMaterialGlobalAssetId(materialItemStock.getMaterial().getMaterialNumberCx());
+        samm.setMaterialNumberCustomer(materialItemStock.getMaterial().getOwnMaterialNumber());
+        samm.setMaterialNumberSupplier(mprService.find(materialItemStock.getMaterial(),
+            materialItemStock.getPartner()).getPartnerMaterialNumber());
+        var posList = new ArrayList<Position>();
+        samm.setPositions(posList);
+        Position position = new Position();
```

Comment:
the posList is not set to the samm. Therefore the position remains empty.

Reply:
Good finding, thanks :)